google-cloud-pubsub/lib/google/cloud/pubsub/service.rb

```diff
@@ -38,65 +36,58 @@ class Service
 
         ##
         # Creates a new Service instance.
-        def initialize project, credentials, host: nil, timeout: nil,
-                       client_config: nil
+        def initialize project, credentials, host: nil, timeout: nil
           @project = project
           @credentials = credentials
-          @host = host || V1::PublisherClient::SERVICE_ADDRESS
+          @host = host
           @timeout = timeout
-          @client_config = client_config || {}
           @client_id = SecureRandom.uuid.freeze
         end
 
-        def channel
-          require "grpc"
-          GRPC::Core::Channel.new host, chan_args, chan_creds
-        end
-
-        def chan_args
-          { "grpc.max_send_message_length"           => -1,
-            "grpc.max_receive_message_length"        => -1,
-            "grpc.keepalive_time_ms"                 => 300_000,
-            "grpc.service_config_disable_resolution" => 1 }
```

Comment:
@dazuma Do we still need to set this to disable the gRPC service config? See PR #3443 and googleapis/gapic-generator#2778.

Reply:
Yes we still need to set this. (Indeed it is set by default: https://github.com/googleapis/google-cloud-ruby/blob/master/google-cloud-pubsub-v1/lib/google/cloud/pubsub/v1/publisher/client.rb#L973)

Reply:
Right! OK, I'll just set the other 3 channel args that I mistakenly removed in the same Config default value.

Reply:
Should we set `max_send_message_length`, `max_receive_message_length` and `keepalive_time_ms` in google-cloud-pubsub-v1 somehow, with the idea being that all users of the pubsub-v1 client need these settings?

Reply:
Hmm. That's a good point. If these should be applied to all users of pubsub-v1, then they really should be set somehow in the generated client.

I don't think this requires a generator change. You can update the default config values in a partial gapic helper.

Reply:
This is fixed in #7054